### PR TITLE
import sound system from Crispy Doom

### PR DESCRIPTION
**Pull request by rfomin**

### CODE REVIEW

File: Source/s_sound.c [hunk 1]
```diff
@@ -794,6 +813,24 @@ void S_Init(int sfxVolume, int musicVolume)
    mus_paused = 0;
 }
 
+void S_ChangeMusicDevice(void)
```
Comment (rfomin) [inline Source/s_sound.c:816]: I made an attempt to change the MIDI synth in real-time. It works on Windows, but I'm not sure about this approach.

**Comment by rfomin** [comment]
I have some questions/suggestions:
1. Personally, I don't remember how a real OPL soundcard sounds. So I think we could remove the authentic OPL emulation and replace it with libADLMIDI, which sounds much more interesting. The downside is the more complicated build process.
2. The original MBF has no PC speaker sound. And what to do with DEHEXTRA sounds? [HID.wad ](https://www.doomworld.com/forum/topic/123082-heretic-invades-doom-two-small-maps-for-mbf21-ports/) does not currently work with PC Speaker.
3. I think we need to get rid of the midiproc. I am planning to use `midifile.c` and change the volume of each MIDI note like GZDoom.
4. I believe we should make it easier for Windows users to access FluidSynth. How about https://github.com/Doom64/fluidsynth-lite or https://github.com/divideconcept/FluidLite?

The alternative to all of this is [SDL Mixer X](https://github.com/WohlSoft/SDL-Mixer-X), which does all of the above.

**Comment by fabiangreffrath** [comment]
Erm, wow, slowly! 😁 

So, if we are not going to use OPL or PC Speaker emulation from Crispy nor midiproc - what exactly do we achieve here?

**Comment by rfomin** [comment]
We can use general code structure, just change/remove some "sound devices". In the case of SDL Mixer X we can preserve the current Woof code, but it's a significant dependency.

**Comment by fabiangreffrath** [comment]
Sure, but that's what I meant. If we are not going to use any of the "sound devices" beyond SDL for music and sounds, we don't actually win anything. Honestly, I have never heard of Mixer-X before, but I'd rather stick with "standard" dependencies.

**Comment by rfomin** [comment]
Well, SDL Mixer X is from mainline SDL_mixer contributor, and they plan to merge many of its features, but I agree that's too much. I love that Woof is remarkably easy to build now.

I suggest that we first merge Crispy's current sound system as it is, and then remove midiproc. We can postpone the rest.

**Comment by fabiangreffrath** [comment]
I question why we would want to merge Crispy's sound system if we are not going to use any of its variable sound backend features besides the on we are already using now?

**Comment by rfomin** [comment]
No, we will be using OPL emulation and PC speaker. I guess these were ideas for future developments :wink:

**Comment by fabiangreffrath** [comment]
Ah, I see! Let's just get the current changes into a 6.2.1 or 6.3 release first.

**Comment by rfomin** [comment]
Another sound device that we could add is caption mode like in Minecraft: [[doomworld]](https://www.doomworld.com/forum/topic/123637-accessibility-in-source-ports/?do=findComment&comment=2360369) Is it feasible?

**Comment by fabiangreffrath** [comment]
I'd even say it would be feabile to have this as a PR in Choco, so that all derived ports (which are a lot) could benefit from it.

**Comment by rfomin** [comment]
I like this idea! One of the problems is the DEHACKED/DEHEXTRA monsters, but I think we can ignore this at first and make patches for autoload.

**Comment by fabiangreffrath** [comment]
I guess it would be alright if we just called unknown monsters just "Monster" or even "Monster #<doomednum>" in the interface.

**Comment by rfomin** [comment]
But how do we know if a monster has been modified? For example, in Hacx Cacodemon is a decoration. But I think in 90% WADs this is not a problem.

**Comment by rfomin** [comment]
Perhaps now we don't need a full-fledged Crispy sound system. The only thing I want to redo is the music API. Also caption mode "device" is probably not a good idea because it forces the sound effects to be disabled. I think it will be better if we have a caption mode and sound at the same time.

**Pull request closed without merging**